### PR TITLE
Update dmodel ioformat

### PR DESCRIPTION
**Pull request by jedwards4b**
Update IO_format for data models, done in a backward compatible way so that externals can continue to use older method (but with io_format hard coded to 64bit_offset)

Test suite: scripts_regression_tests.py
Test baseline: 
Test namelist changes: 
Test status: bit for bit

Fixes 

User interface changes?: 

Update gh-pages html (Y/N)?:

Code review:

### CODE REVIEW

**Comment by jedwards4b** [comment]
Reran scripts_regression_tests.py - all pass.

File: config/cesm/machines/config_batch.xml [hunk 1]
```diff
@@ -397,6 +397,7 @@
     <batch_submit>sbatch</batch_submit>
     <directives>
       <directive>-C knl,quad,cache </directive>
+      <directive>-S 2 </directive>
```
Comment (jayeshkrishna) [inline config/cesm/machines/config_batch.xml]: This change should be in a different PR

File: config/xml_schemas/env_mach_specific.xsd [hunk 2]
```diff
@@ -4,7 +4,7 @@
 <xs:attribute name="id" type="xs:NCName"/>
 <xs:attribute name="version" type="xs:decimal"/>
 <xs:attribute name="lang" type="xs:NCName"/>
-<xs:attribute name="compiler" type="xs:NCName"/>
+<xs:attribute name="compiler" type="xs:string"/>
```
Comment (jayeshkrishna) [inline config/xml_schemas/env_mach_specific.xsd]: This change should also be in a separate PR.
